packages/connection-manager/src/js/ConnectionQueue/__tests__/ConnectionQueue-test.js

```diff
@@ -0,0 +1,210 @@
+import { default as Connection } from "../../Connection/AbstractConnection";
+import ConnectionQueue from "../ConnectionQueue.js";
+// import ConnectionQueueItem from "../ConnectionQueueItem.js";
+
+jest.mock("../../Connection/AbstractConnection");
+// jest.mock("../ConnectionQueueItem");
+
+/* per default, jest reads /__mocks__/immutable.js even with automock disabled. */
+jest.unmock("immutable");
+
+// Object.assign(global[Symbol.for("$$jest-matchers-object")].matchers, {
+//   toBeMagic(received, predicate) {
+//     console.log("received mock", received.mock);
+
+//     return {
+//       message: "Predicate function not have been called",
+//       pass: received.mock.calls.some(predicate)
+//     };
+//   }
+// });
+
+describe("ConnectionQueue", () => {
+  describe("#init", () => {
+    it("initializes successfully", () => {
+      expect(() => new ConnectionQueue()).not.toThrow();
+    });
+    it("has correct size after initialization (head)", () => {
+      expect(new ConnectionQueue().size).toEqual(0);
+    });
+    it("is empty after initialization (size)", () => {
+      expect(new ConnectionQueue().size).toEqual(0);
+    });
+    it("throws on initialization with invalid list", () => {
+      expect(() => new ConnectionQueue("foobar")).toThrow();
+    });
+  });
+  describe("#enqueue", () => {
+    it("enqueues correctly (head)", () => {
+      const connection = new Connection("asd");
+      const queue = new ConnectionQueue().enqueue(connection);
+
+      expect(queue.head()).toEqual(connection);
+    });
+    it("enqueues correctly (size)", () => {
+      const queue = new ConnectionQueue().enqueue(new Connection("asd"));
+
+      expect(queue.size).toEqual(1);
+    });
+  });
+  describe("#dequeue", () => {
+    it("enqueues correctly (head)", () => {
+      const connection = new Connection("asd");
+      const queue = new ConnectionQueue()
+        .enqueue(connection)
+        .dequeue(connection);
+
+      expect(() => queue.head()).toThrow();
+    });
+    it("enqueues correctly (size)", () => {
+      const connection = new Connection("asd");
+      const queue = new ConnectionQueue()
+        .enqueue(connection)
+        .dequeue(connection);
+
+      expect(queue.size).toEqual(0);
+    });
+  });
+  describe("#contains", () => {});
+});
+
+/*
+var ConnectionStore;
```

Comment:
These test seem old. Please remove the old test and add a sufficient set for the `ConnectionQueue`